s2cloudless/pixel_classifier.py

```diff
@@ -46,56 +45,37 @@ def _check_classifier(classifier: Any) -> None:
             if not callable(method):
                 raise ValueError(f"Classifier does not have a {method_name} method!")
 
-    @staticmethod
-    def extract_pixels(data: np.ndarray) -> np.ndarray:
-        """Extracts pixels from data array
-
-        :param data: Array of images to be classified.
-        :return: Reshaped 2D array
-        :raises: ValueError is input array has wrong dimensions
-        """
-        if len(data.shape) != 4:
-            raise ValueError(
-                "Array of input images has to be a 4-dimensional array of shape"
-                "[n_images, n_pixels_y, n_pixels_x, n_bands]"
-            )
-
-        return data.reshape((-1, data.shape[3]))
-
     def image_predict(self, data: np.ndarray, **kwargs: Any) -> np.ndarray:
-        """
-        Predicts class labels for the entire image.
+        """Predicts class labels for the entire image.
 
-        :param data: Array of images to be classified.
+        :param data: Array of images of shape `(N, height, width, bands)` to be classified.
         :param kwargs: Any keyword arguments that will be passed to the classifier's prediction method
-        :return: raster classification map
+        :return: Raster classification map of shape `(N, height, width)`
         """
-        pixels = self.extract_pixels(data)
-
         if isinstance(self.classifier, Booster):
             raise NotImplementedError(
-                "An instance of lightgbm.Booster can only return prediction probabilities, "
-                "use PixelClassifier.image_predict_proba instead"
+                "An instance of `lightgbm.Booster` can only return prediction probabilities, use the"
+                "`PixelClassifier.image_predict_proba` instead."
             )
 
+        pixels = data.reshape((-1, data.shape[-1]))
         predictions = self.classifier.predict(pixels, **kwargs)
 
-        return predictions.reshape(*data.shape[0:3])
+        return predictions.reshape(*data.shape[:-1])
 
     def image_predict_proba(self, data: np.ndarray, **kwargs: Any) -> np.ndarray:
-        """
-        Predicts class probabilities for the entire image.
+        """Predicts class probabilities for the entire image.
 
-        :param data: Array of images to be classified.
+        :param data: Array of images of shape `(N, height, width, bands)` to be classified.
         :param kwargs: Any keyword arguments that will be passed to the classifier's prediction method
-        :return: classification probability map
+        :return: Classification probability map of shape `(N, height, width, bands)`
```

Comment:
i noticed that the docstring here is wrong. Please check what the shape of the output is, i suspect it is `(N, height, width, 1)`

Reply:
It is used in `get_cloud_probability_maps` as `proba = self.classifier.image_predict_proba(data, **kwargs)[..., 1]`, from existing code i would guess that docstring is "correct ".

Reply:
ah, it could be that it is of size 2, probability of 0 and probability of 1.
The `bands` is wrong, because it has no relation to bands. Perhaps `classes`?

Reply:
incorrect * should be num of posible class `probabilities.shape[-1]`

Reply:
updated